client/thin-replica-client/include/client/thin-replica-client/grpc_connection.hpp

```diff
@@ -77,17 +78,29 @@ class GrpcConnection {
   // Possible results of RPC operations a GrpcConnection may report.
   enum class Result { kUnknown, kSuccess, kFailure, kTimeout, kOutOfRange, kNotFound, kEndOfStream };
 
+  static const size_t defaultSubscriptionThreadPoolSize = 1;
```

Comment:
@arc-vmware so in general will eliminate all the thread pool config variables and bootstrap with pool size of one. As you mentioned we can later add config via grpcconfig and tune as required. Does this sound good?

Reply:
@bsenthilr yes i was saying the following:
if(thread_pool_size is available) {
    subscription_pool_ = std::make_unique<concord::util::ThreadPool>(*thread_pool_size);
} else {
    subscription_pool_ = std::make_unique<concord::util::ThreadPool>();
}
This will be done in the ctor.

Reply:
@arc-vmware this capability was leveraged using single thread on pool by default. This PR is regressed (performance tested) with single thread on pool as of now. I can explicitly make it create pool on one thread for now. Our future tuning and updates can bring in setting the size via configuration. The default constructor initializes pool of maximum supported thread that is auto calculated. We can adjust this code after more testing as requrid.